### PR TITLE
Deprecate types in update_by_query and delete_by_query

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Relates to #35190

### CODE REVIEW

**Review by jtibshirani — COMMENTED** [review]
Thanks @mayya-sharipova, this looks good to me overall! I left some small comments, and also noted that it would be good to use the changes in #36328.

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java [hunk 1]
```diff
@@ -19,23 +19,61 @@
...
 import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.usage.UsageService;
 
 import java.io.IOException;
+import java.util.Collections;
 
 import static java.util.Collections.emptyList;
 import static org.mockito.Mockito.mock;
 
 public class RestDeleteByQueryActionTests extends ESTestCase {
```
[the remark this answers is not in the document]
Comment (mayya-sharipova) [reply modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java]: @jtibshirani  Thanks Julie.  What "request object" do you have in mind here?  Is it `RestRequest`?  Does this line introduced in my last commit is enough for testing:
```
assertEquals("some_type", request.param("type"));
```

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Comment by mayya-sharipova** [comment]
@jtibshirani  Thanks for the review. I have tried to address your comments. Can you please continue the review when you have time

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java [hunk 2]
```diff
@@ -21,20 +21,39 @@
...
+    }
+
+    public void testTypeInPath() {
+        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
+            .withMethod(RestRequest.Method.POST)
+            .withPath("/some_index/some_type/_delete_by_query")
+            .build();
+        dispatchRequest(request);
+        // checks the type in the URL is propagated correctly to the request object
+        assertEquals("some_type", request.param("type"));
```
Comment (jtibshirani) [inline modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java]: I was thinking we could have a separate test for request building where we call `action.buildRequest(...)` and check that the `DeleteByQueryRequest` contains the right document types.

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtibshirani) [reply modules/reindex/src/test/java/org/elasticsearch/index/reindex/RestDeleteByQueryActionTests.java]: I was thinking we could have a separate test for request building where we call `action.buildRequest(...)` and check that the return value (like `DeleteByQueryRequest`) contains the right document types.

**Review by jtibshirani — APPROVED** [review]
Looks good to me.

**Pull request merged**